pdl-derive/src/lib.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use codespan_reporting::term;
+use proc_macro2::TokenStream;
+use quote::quote;
+use std::env;
+use std::path::Path;
+use syn::parse_macro_input;
+
+fn pdl_proc_macro(path: syn::LitStr, input: syn::ItemMod) -> TokenStream {
+    // Locate the source grammar file.
+    let root = env::var("CARGO_MANIFEST_DIR").unwrap_or_else(|_| ".".into());
+    let Some(relative_path) =
+        [Path::new(&root).join(path.value()), Path::new(&root).join("src").join(path.value())]
```

Comment:
I'm confused about how this works when using it in code — perhaps we could add an `example/` to show this?

What confuses me is the literal `src` directory here: what if I don't use such a folder? Now that I look closer, I see that you're _searching_ for a PDL file. Why do we need to search? Should the macro not use an unambiguous path?

Reply:
Typically you would use the relative path from the Cargo.toml file.
The sub-path  `"src"` is added as additional option.
Would you rather only support absolute paths from the cargo root ?

Reply:
> Typically you would use the relative path from the Cargo.toml file.

I would have expected the path to be relatively to the source file where the macro is found — I believe this is how `include_str!` and related macros work. I don't know if that's possible, though?

Reply:
> Would you rather only support absolute paths from the cargo root ?

Yes, I would suggest removing convenience features such as searching in a `src/` directory by default. It's simpler to document and it's simpler to understand :slightly_smiling_face: 

> But I agree with you, it would be better to be consistent with `include_str!`, I created #63 to track this issue.

Thanks and thanks for the explanation!